merkle/sparse_merkle_tree.go

```diff
@@ -215,14 +229,19 @@ func (s *subtreeWriter) buildSubtree() {
 				return nil, nil
 			}
 			if expected, got := nodeID, nodes[0].NodeID; !expected.Equivalent(got) {
-				return nil, fmt.Errorf("expected node ID %s from storage, but got %s", expected, got)
+				return nil, fmt.Errorf("expected node ID %s from storage, but got %s", expected.String(), got.String())
```

Comment:
Yes, the formatter doesn't know to call it on your behalf (though I think you can (somehow) do that - big.Int seems to do that.)